projects/bp-gallery/src/components/common/PictureOverview.tsx

```diff
@@ -24,18 +26,52 @@ const PictureOverview = ({
   onClick,
   sortBy,
   rows = 2,
+  type = PictureOverviewType.CUSTOM,
 }: PictureOverviewProps) => {
   const { t } = useTranslation();
+  const { role } = useAuth();
 
-  const { data, loading, refetch } = useGetPictures(
-    queryParams,
-    false,
-    sortBy,
-    true,
-    ABSOLUTE_MAX_PICTURES_PER_ROW * rows
-  );
+  const { getPicturesQuery } = useGenericPictureEndpoints(type);
+
+  const textFilter = useMemo(() => {
+    return role < AuthRole.CURATOR
+      ? {
+          and: [
+            {
+              or: [
+                {
+                  is_text: {
+                    eq: false,
+                  },
+                },
+                {
+                  is_text: {
+                    null: true,
+                  },
+                },
+              ],
+            },
+            queryParams as PictureFiltersInput,
+          ],
+        }
+      : queryParams;
+  }, [queryParams, role]);
+
+  const { data, loading, refetch } = getPicturesQuery({
+    variables: {
+      filters: textFilter as PictureFiltersInput,
+      pagination: {
+        start: 0,
+        limit: ABSOLUTE_MAX_PICTURES_PER_ROW * rows,
+      },
+      sortBy,
+    },
+    fetchPolicy: 'network-only',
+  });
```

Comment:
Maybe put the `useGetMostLikedPicturesQuery` inside `useGetPictures` instead of copy-pasting the code here?